################################################################################
SHORT DESCRIPTION: 
################################################################################
Port git://xenbits.xen.org/xen.git:a9378c3785df4bc8a5babb004c69a503c3830993

################################################################################
LONG DESCRIPTION: 
################################################################################
commit a9378c3785df4bc8a5babb004c69a503c3830993
Author: Andrew Cooper <andrew.cooper3@citrix.com>
Date:   Mon Nov 25 11:04:21 2013 +0000

    firmware/vgabios: Fix incorrect formatting string
    
    bios_len is an long, so requires %ld rather than %d.
    
    Coverity ID: 1055813
    
    Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
    Acked-by: Ian Campbell <Ian.Campbell@citrix.com>
    CC: Keir Fraser <keir@xen.org>
    CC: Jan Beulich <JBeulich@suse.com>
    CC: Ian Jackson <Ian.Jackson@eu.citrix.com>

################################################################################
CHANGELOG 
################################################################################
Ported to VGABIOS: Eric Chanudet, chanudete@ainfosec.com, 01/03/2015

################################################################################
REMOVAL 
################################################################################
This patch can be removed if a new version of Bochs VGABIOS is released with
this fix.

################################################################################
UPSTREAM PLAN 
################################################################################
This is already upstream in Xen repository.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None.

################################################################################
PATCHES
################################################################################
--- a/biossums.c
+++ b/biossums.c
@@ -161,7 +161,7 @@ int main(int argc, char* argv[])
       exit(EXIT_FAILURE);
     }
     if (fwrite(bios_data, 1, bios_len, stream) < bios_len) {
-      printf("Error writing %d KBytes to %s.\n", bios_len / 1024, argv[1]);
+      printf("Error writing %ld KBytes to %s.\n", bios_len / 1024, argv[1]);
       fclose(stream);
       exit(EXIT_FAILURE);
     }
